test/unit/gateways/cyber_source_rest_test.rb

```diff
@@ -233,6 +223,46 @@ def test_url_building
     assert_equal "#{@gateway.class.test_url}/pts/v2/action", @gateway.send(:url, 'action')
   end
 
+  def test_successful_store
+    stub_comms(@gateway, :ssl_request) do
+      @gateway.store(@credit_card, @options.merge(merchant_customer_id: 'merchant_test'))
+    end.check_request do |_method, endpoint, data, _headers|
+      request = JSON.parse(data)
+      case endpoint
+      when /instrumentidentifiers/
+        assert_equal request['card']['number'], '4111111111111111'
+      when /paymentinstruments/
+        assert_equal request['card']['expirationMonth'], '12'
+        assert_equal request['card']['expirationYear'],  '2031'
+        assert_equal request['card']['type'], 'visa'
+        assert_equal request['instrumentIdentifier']['id'], '7010000000016241111'
+        assert_includes request, 'billTo'
+      when /payment-instruments/
+        assert_equal request['card']['expirationMonth'], '12'
+        assert_equal request['card']['expirationYear'],  '2031'
+        assert_equal request['card']['type'], '001'
+        assert_equal request['instrumentIdentifier']['id'], '7010000000016241111'
+        assert_includes request, 'billTo'
+      when /customers/
+        assert_equal request['buyerInformation']['email'], 'test@cybs.com'
+        assert_includes request['buyerInformation'], 'merchantCustomerId'
+        assert_includes request, 'clientReferenceInformation'
+        assert_includes request, 'merchantDefinedInformation'
+      end
+    end.respond_with(successful_create_customer, successful_create_instrument_identifiers, successful_store_response)
+  end
```

Comment:
A couple tests I think we should try:

- If the instrument creation call fails, does the authroization string still contain a customer_id from the first step? Otherwise we have an "orphan" customer.
- Similar, for instrument identifier, if the last call fails - but we're currently not returning the instrument id in the authorization string, so we should add that if possible.

Reply:
the case with the payment instrument creation failure was added, the final case was tried but not able to inject failure due to previous calls guarantee the success of this statement.